src/Http/Router.php

```diff
@@ -45,44 +36,32 @@ public function __construct(RouteParser $routeParser, DataGenerator $dataGenerat
      */
     public function addRoute(Route $route): void
     {
-        if (!$route instanceof Route) {
-            throw RouterException::invalidRoute($route);
-        }
-        $routeDatas = $this->routeParser->parse($route->getExpression());
-        foreach ((array) $route->getMethod() as $method) {
-            foreach ($routeDatas as $routeData) {
-                $this->dataGenerator->addRoute($method, $routeData, $route);
-            }
-        }
+        $baseRoute = $route->getBaseRoute();
+        $this->routeCollection->add($route->getName(), $baseRoute);
+        $this->routes[$route->getName()] = $route;
     }
 
     /**
      * Finds route matching clients request.
      *
      * @param string $method request method.
-     * @param string $uri request uri.
+     * @param string $path request path.
      * @return Route
      */
-    public function findRoute(string $method, string $uri): Route
+    public function findRoute(string $method, string $path): Route
     {
-        $dispatcher = new GroupCountBased($this->getData());
-        $info = $dispatcher->dispatch($method, $uri);
-
-        switch ($info[0]) {
-            case Dispatcher::NOT_FOUND:
-                throw NotFoundException::notFound($uri, $method);
-
-            case Dispatcher::METHOD_NOT_ALLOWED:
-                $allowedMethods = $info[1];
-                throw MethodNotAllowedException::methodNotAllowed($uri, $method, $allowedMethods);
-
-            case Dispatcher::FOUND:
-                return $info[1]->withAttributes($info[2]);
+        $matcher = new UrlMatcher($this->routeCollection, new RequestContext('/', $method));
+        try {
+            $route = $matcher->match($path);
+        } catch (ResourceNotFoundException $exception) {
+            throw NotFoundException::notFound($path, $method);
+        } catch (MethodNotAllowedException $exception) {
+            throw \Igni\Http\Exception\MethodNotAllowedException::methodNotAllowed($path, $method, $exception->getAllowedMethods());
```

Comment:
I think it's better to import your own and keep FQDN for 3rd party; or even better - use alias like `SymfonyMethodNotAllowedException`